test/moxinet/plug/mocked_response_test.exs

```diff
@@ -34,6 +34,26 @@ defmodule Moxinet.Plug.MockedResponseTest do
       assert conn.resp_body == Jason.encode!(response_body)
     end
 
+    test "Sends non nil payload for non 'application/json' post requests" do
+      test_pid = Kernel.self()
+      _ = SignatureStorage.start_link(name: SignatureStorage)
+      response_body = "test=yes"
+
+      conn =
+        conn(:post, "/path")
+        |> put_req_header("x-moxinet-ref", Moxinet.pid_reference(self()))
+
+      SignatureStorage.store(CustomAPIMock, :post, "/path", fn payload ->
+        Kernel.send(test_pid, {:post_payload, payload})
+        %{status: 200, body: response_body}
```

Comment:
```suggestion
      response_body = "test=yes"

      conn =
        conn(:post, "/path")
        |> put_req_header("x-moxinet-ref", Moxinet.pid_reference(self()))

      SignatureStorage.store(CustomAPIMock, :post, "/path", fn payload ->
        Kernel.send(test_pid, {:post_payload, payload})
        %{status: 200, body: "test=yes"}
```

Since we're not verifying the body anywhere in our assertions we can define it inline to not give a sense of it being more important than it is for this test